src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackReviews/FeedbackReviews.tsx

```diff
@@ -1,127 +1,64 @@
+
   const [assessmentStep, setAssessmentStep] = useState(0);
   const [submitAssessmentLoading, setSubmitAssessmentLoading] = useState(false);
   const [submitAssessmentError, setSubmitAssessmentError] = useState<string | null>(null);
   const [showFeedbackAssessment, setShowFeedbackAssessment] = useState(false);
 
-  const { reset } = useFormContext<FeedbackForm>();
   const { user } = auth.useData() ?? {};
+
   const {
-    assessment,
-    lastAssessment,
-    setAssessment,
-    setIsLastVersion,
     isBannerVisible,
     setIsBannerVisible,
-  } = useContext(RespondentDataReviewContext);
-  const getFeedbackReviewsData = useFeedbackReviewsData();
+    reviewerData,
+    reviewsLoading,
+    reviewsFlowLoading,
+    reviewsError,
+    reviewsFlowsError,
+    removeReviewsLoading,
+    removeReviewsFlowLoading,
+    removeReviewsError,
+    removeReviewsFlowError,
+    handleGetReviews,
+    handleReviewerAnswersRemove,
+  } = useFeedbackReviews({ appletId, answerId, submitId, user, setAssessmentStep });
 
   const userName = `${user?.firstName ?? ''} ${user?.lastName ?? ''}${t('me')}`;
   const dataTestid = 'respondents-data-summary-feedback-reviewed';
 
-  const {
-    execute: getReviews,
-    isLoading: reviewsLoading,
-    error: reviewsError,
-  } = useAsync(getReviewsApi, async (result) => {
-    const reviewsData = await getFeedbackReviewsData({
-      reviews: result?.data?.result ?? [],
-      userId: user?.id ?? '',
-    });
-
-    setReviewersData(reviewsData);
-  });
-
-  const {
-    execute: removeReview,
-    isLoading: removeReviewLoading,
-    error: removeReviewError,
-  } = useAsync(deleteReviewApi);
-
-  const updateAssessment = (updatedAssessment: AssessmentActivityItem[]) => {
-    setAssessment(updatedAssessment);
-    reset(getDefaultFormValues(updatedAssessment));
-    setAssessmentStep(0);
-  };
-
-  const handleSelectLastVersion = () => {
-    setIsLastVersion(true);
-    setIsBannerVisible(false);
-
-    const updatedAssessment = lastAssessment?.map((activityItem) => ({
-      activityItem,
-      answer: undefined,
-    })) as AssessmentActivityItem[];
-    updateAssessment(updatedAssessment);
-  };
-
-  const handleAssessmentBannerClose = () => setIsBannerVisible(false);
-
-  const handleReviewerAnswersRemove = async ({ assessmentId }: AssessmentId) => {
-    if (!appletId || !answerId) return;
-    await removeReview({ appletId, answerId, assessmentId });
-
-    if (lastAssessment?.length) {
-      handleSelectLastVersion();
-    } else {
-      const updatedAssessment = assessment?.map(({ activityItem, items }) => ({
-        activityItem,
-        answer: undefined,
-        items,
-      })) as AssessmentActivityItem[];
-      updateAssessment(updatedAssessment);
-    }
-    getReviews({ appletId, answerId });
-  };
-
   const handleReviewEdit = () => {
     setShowFeedbackAssessment(true);
   };
 
-  const hasCurrentUserReview = reviewersData.some(
+  const hasCurrentUserReview = reviewerData.some(
```

Comment:
Maybe it's better to keep `reviewers` plural since it represents data for multiple reviewers, not just a single reviewer?